storage/badger/transaction_results.go

```diff
@@ -16,37 +16,32 @@ type TransactionResults struct {
 	cache *Cache
 }
 
-type BlockIDTransactionID struct {
-	BlockID       flow.Identifier
-	TransactionID flow.Identifier
+func KeyFromBlockIDTransactionID(blockID flow.Identifier, txID flow.Identifier) []byte {
+	var key [64]byte
+	copy(key[0:32], blockID[:])
+	copy(key[32:64], txID[:])
+	return key[:]
 }
 
-func KeyFromBlockIDTransactionID(blockID flow.Identifier, txID flow.Identifier) string {
-	return fmt.Sprintf("%x%x", blockID, txID)
-}
-
-func KeyToBlockIDTransactionID(key string) (flow.Identifier, flow.Identifier, error) {
-	blockIDStr := key[:64]
-	txIDStr := key[64:]
-	blockID, err := flow.HexStringToIdentifier(blockIDStr)
-	if err != nil {
-		return flow.ZeroID, flow.ZeroID, fmt.Errorf("could not get block ID: %w", err)
+func KeyToBlockIDTransactionID(key []byte) (flow.Identifier, flow.Identifier, error) {
+	if len(key) != 64 {
+		return flow.ZeroID, flow.ZeroID, fmt.Errorf("could not convert key to block id and tx id, key size: %v", len(key))
 	}
 
-	txID, err := flow.HexStringToIdentifier(txIDStr)
-	if err != nil {
-		return flow.ZeroID, flow.ZeroID, fmt.Errorf("could not get transaction id: %w", err)
-	}
+	var blockID [32]byte
```

Comment:
Guess we don't save much here by switching to bytes, I assumed we had something like `flow.IdentifierFromBytes`

Reply:
we don't that method